src/Datadog.Trace/Tagging/ITags.cs

```diff
@@ -1,7 +1,13 @@
+using System.Collections.Generic;
+
 namespace Datadog.Trace.Tagging
 {
     internal interface ITags
     {
+        List<KeyValuePair<string, double>> Metrics { get; }
+
+        List<KeyValuePair<string, string>> Tags { get; }
```

Comment:
I opted to add this and simply expose this member from the abstract class but perhaps we can refine this, right now the naming is a bit awkward: `span.Tags.Tags`. That said I think that is acceptable for this change.

Reply:
Would it read better if it was something like `span.Tags.All` or `span.Tags.AllTags`?

Reply:
For more context, we created this ITags interface and classes to optimize the common case performance where no modifications are made by the user to automatic instrumentation tags. That's also why the this `ITags` interface unintuitively has a member for both tags and metrics. This change is definitely good, though the naming is still awkward because of the original naming choices